File: src/main/java/hu/frontrider/arcana/recipes/formulacrafting/FormulaRecipes.java
package hu.frontrider.arcana.recipes.formulacrafting;

import net.minecraft.block.Block;

import java.util.*;

public class FormulaRecipes {
    public static FormulaRecipes INSTANCE;

    Map<Block,List<FormulaApplicationRecipe>> recipes;

    public static void create(){
        INSTANCE = new FormulaRecipes();
    }

    private FormulaRecipes(){
        recipes = new LinkedHashMap<>();
    }

    public void addRecipe(FormulaApplicationRecipe recipe)
    {
        if(recipes.containsKey(recipe.getBlock())) {
            recipes.get(recipe.getBlock()).add(recipe);
        }
        else
        {
            List<FormulaApplicationRecipe> formulaApplicationRecipeArrayList = new LinkedList<>();
            formulaApplicationRecipeArrayList.add(recipe);
            recipes.put(recipe.getBlock(),formulaApplicationRecipeArrayList);
        }
    }

    public Collection<List<FormulaApplicationRecipe>> getRecipes() {
        return recipes.values();
    }
}

File: src/main/java/hu/frontrider/arcana/recipes/formulacrafting/FormulaRecipes.kt
package hu.frontrider.arcana.recipes.formulacrafting

import net.minecraft.block.Block

import java.util.*

object FormulaRecipes {

    val recipes: MutableMap<Block, List<FormulaApplicationRecipe>> = LinkedHashMap()

    fun addRecipe(recipe: FormulaApplicationRecipe) {
        if (recipes.containsKey(recipe.block)) {
            recipes[recipe.block]!!.plus(recipe)
        } else {
            val formulaApplicationRecipeArrayList = LinkedList<FormulaApplicationRecipe>()
            formulaApplicationRecipeArrayList.add(recipe)
            recipes[recipe.block] = formulaApplicationRecipeArrayList
        }
    }

    fun getRecipesCollection(): Collection<List<FormulaApplicationRecipe>> {
        return recipes.values
    }

    fun init(){

    }

}
